add py3-ansible-core package

kranurag7 opened the pull request:
add py3-ansible package

notes around why we are building with a patch:

        this support is required because we have latest resolvlib and that's
        supported on the upstream project devel branch too.
        the tag we are checking out is using older version of resolvlib and
        using that is leading to build failure unless we use the old version of
        resolvlib which we don't have packaged in wolfi.

kranurag7 committed:
add py3-ansible package

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

kranurag7 committed:
change runtime deps consider provider-priority

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

EyeCantCU commented on py3-ansible.yaml:
```diff
@@ -0,0 +1,75 @@
+    strip-prefix: v
+
+test:
+  environment:
+    contents:
+      packages:
+        - py3-yaml
+        - py3-jinja2
```
No need to redefine package runtime dependencies

EyeCantCU commented on py3-ansible.yaml:
```diff
@@ -0,0 +1,75 @@
+    identifier: ansible/ansible
+    strip-prefix: v
+
+test:
```
We should add an import test as well. See other versioned Python packages as examples

EyeCantCU commented on py3-ansible.yaml:
```diff
@@ -0,0 +1,75 @@
+    provider-priority: 0
+    runtime:
+      - py3-jinja2
+      - py3-yaml
```
No need to define here, the empty main package will never be used

EyeCantCU commented on py3-ansible.yaml:
```diff
@@ -0,0 +1,75 @@
+        - py3-${{vars.pypi-package}}
+    pipeline:
+      - name: Python Build
+        uses: python/build-wheel
```
Should probably be `py/pip-build-install`

kranurag7 replied:
RJ, when I try to use this pipeline then I'm not able to build for python-3.10.

```bash
2024/11/08 12:47:49 WARN   creating '/tmp/pip-modern-metadata-cbsh28yi/ansible_core-2.18.0.dist-info' subpackage=py3.10-ansible-core uses=py/pip-build-install name="pip build"
2024/11/08 12:47:49 INFO   Preparing metadata (pyproject.toml): finished with status 'done' subpackage=py3.10-ansible-core uses=py/pip-build-install name="pip build"
2024/11/08 12:47:49 INFO INFO: pip is looking at multiple versions of ansible-core to determine which version is compatible with other requirements. This could take a while. subpackage=py3.10-ansible-core uses=py/pip-build-install name="pip build"
2024/11/08 12:47:49 WARN ERROR: Package 'ansible-core' requires a different Python: 3.10.15 not in '>=3.11' subpackage=py3.10-ansible-core uses=py/pip-build-install name="pip build"
```

EyeCantCU replied:
We can't build with 3.10 as it needs at least 3.11 so let's remove 3.10

EyeCantCU commented on py3-ansible.yaml:
```diff
@@ -0,0 +1,75 @@
+    description: python${{range.key}} version of ${{vars.pypi-package}}
+    dependencies:
+      runtime:
+        - py3-jinja2
```
Should always be `py${{range.key}}-*` for versioned deps

kranurag7 committed:
change name to ansible-core and multi-versioned runtime deps

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

EyeCantCU commented on py3-ansible-core.yaml:
```diff
@@ -0,0 +1,74 @@
+    strip-prefix: v
+
+test:
+  environment:
+    contents:
+      packages:
+        - py3-jinja2
+        - py3-pyyaml
```
We can drop this

kranurag7 committed:
address review comments

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

kranurag7 committed:
add more runtime deps

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

kranurag7 committed:
add openssl to runtime dependencies

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

kranurag7 committed:
build py3-ansible-core with a patch

this support is required because we have latest resolvlib and that's
supported on the upstream project devel branch too.
the tag we are checking out is using older version of resolvlib and
using that is leading to build failure unless we use the old version of
resolvlib which we don't have packaged in wolfi.

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

maxgio92 commented on py3-ansible-core/resolvlib.patch:
```diff
@@ -0,0 +1,166 @@
++++ b/changelogs/fragments/update-resolvelib-lt-1_2_0.yml
+@@ -0,0 +1,2 @@
++minor_changes:
++  - ansible-galaxy - support ``resolvelib >= 0.5.3, < 1.2.0`` (https://github.com/ansible/ansible/issues/84217).
```
Because I see that it's been patched to `2.0.0` in the upstream and [merged](https://github.com/ansible/ansible/pull/84218/files#diff-065abed58d290df207524c494d40352b8dc2fb11ecb86b3e58553c9a3888656fR15) to the `devel` branch,
even if not setting the update to manual, maybe we should highlight this in case the patch will no longer apply in the next release?

kranurag7 replied:
I think this will be removed as part of automation in the next update.

maxgio92 reviewed (approved):
LGTM

maxgio92 commented:
I think that the filed low CVEs should be marked as false positive. Only `/usr/lib/python3.12/site-packages/ansible_core-2.18.0.dist-info/METADATA` file is there once packages are installed and it's a text file.
